### PR TITLE
fix: Fix toner detection discovery #5391

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

I think this is right.

Fixes: #5391 

Before hand duplicates were detected by capacity_oid which could be null, switched to oid as that will be unique (or should be).

The old if `($os` check was fundamentally wrong as $os is not defined.

The idea being now we use the standard Printer-MIB details and fall back if they are empty which in some situations they are. We also get the internationalised text to use for descr if it's available.

### CODE REVIEW

File: includes/discovery/toner.inc.php [hunk 1]
```diff
@@ -54,11 +66,12 @@
 d_echo("\n Checking valid toner ... \n");
 d_echo($valid_toner);
 
-$sql = "SELECT * FROM toner WHERE device_id = '" . $device['device_id'] . "'";
-foreach (dbFetchRows($sql) as $test_toner) {
-    $toner_index = $test_toner['toner_index'];
+$sql = dbFetchRows("SELECT * FROM toner WHERE device_id = '" . $device['device_id'] . "'");
```
Comment (murrant) [inline includes/discovery/toner.inc.php]: $sql variable name is a bit odd

Comment (laf) [inline includes/discovery/toner.inc.php]: Done

**Comment by murrant** [comment]
Worked on every printer I could test.  But I don't have a wide selection.

**Comment by jquagga** [comment]
This works on my troublesome Samsung ML series as well.  That's all I have to test.

**Pull request merged**